src/mscorlib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -382,19 +382,20 @@ public override void Emit(OpCode opcode, SignatureHelper signature)
         // Exception related generation
         //
         //
-        public override Label BeginExceptionBlock()
+        public override void BeginExceptFilterBlock()
         {
-            return base.BeginExceptionBlock();
-        }
+            // Begins an exception filter block. Emits a branch instruction to the end of the current exception block.
 
-        public override void EndExceptionBlock()
-        {
-            base.EndExceptionBlock();
-        }
+            if (CurrExcStackCount == 0)
+                throw new NotSupportedException(Environment.GetResourceString("Argument_NotInExceptionBlock"));
 
-        public override void BeginExceptFilterBlock()
-        {
-            throw new NotSupportedException(Environment.GetResourceString("InvalidOperation_NotAllowedInDynamicMethod"));
+            __ExceptionInfo current = CurrExcStack[CurrExcStackCount - 1];
+
+            Label endLabel = current.GetEndLabel();
+            Emit(OpCodes.Leave, endLabel);
+            UpdateStackSize(OpCodes.Nop, 1);
```

Comment:
It looks strange to me too, though it seems to work. That said, I've only used that method of the base class with 4.x framework so maybe a bug was introduced and no test covers it. Shall investigate…